services/mediastoredata/src/it/java/software/amazon/awssdk/services/mediastoredata/TransferEncodingChunkedIntegrationTest.java

```diff
@@ -0,0 +1,201 @@
+ */
+
+package software.amazon.awssdk.services.mediastoredata;
+
+import static software.amazon.awssdk.utils.FunctionalUtils.invokeSafely;
+
+import io.reactivex.Flowable;
+import java.io.ByteArrayInputStream;
+import java.io.FilterInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.reactivestreams.Subscriber;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.core.async.AsyncRequestBody;
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.http.ContentStreamProvider;
+import software.amazon.awssdk.http.apache.ApacheHttpClient;
+import software.amazon.awssdk.http.nio.netty.NettyNioAsyncHttpClient;
+import software.amazon.awssdk.http.urlconnection.UrlConnectionHttpClient;
+import software.amazon.awssdk.services.mediastore.MediaStoreClient;
+import software.amazon.awssdk.services.mediastore.model.Container;
+import software.amazon.awssdk.services.mediastore.model.ContainerStatus;
+import software.amazon.awssdk.services.mediastore.model.DescribeContainerResponse;
+import software.amazon.awssdk.services.mediastoredata.model.DeleteObjectRequest;
+import software.amazon.awssdk.services.mediastoredata.model.PutObjectRequest;
+import software.amazon.awssdk.testutils.Waiter;
+import software.amazon.awssdk.testutils.service.AwsIntegrationTestBase;
+
+public class TransferEncodingChunkedIntegrationTest extends AwsIntegrationTestBase {
+    private static final String CONTAINER_NAME = "java-sdk-test-" + Instant.now().toEpochMilli();
+    private static MediaStoreClient mediaStoreClient;
+    private static MediaStoreDataClient syncClientWithApache;
+    private static MediaStoreDataClient syncClientWithUrlConnection;
+    private static MediaStoreDataAsyncClient asyncClientWithNetty;
+    private static AwsCredentialsProvider credentialsProvider;
+    private static Container container;
+    private static PutObjectRequest putObjectRequest;
+    private static DeleteObjectRequest deleteObjectRequest;
+
+    @BeforeAll
+    public static void setup() {
+        credentialsProvider = getCredentialsProvider();
+        mediaStoreClient = MediaStoreClient.builder()
+                                           .credentialsProvider(credentialsProvider)
+                                           .httpClient(ApacheHttpClient.builder().build())
+                                           .build();
+        container = createContainer();
+        URI uri = URI.create(container.endpoint());
+
+        syncClientWithApache = MediaStoreDataClient.builder()
```

Comment:
Can we add an interceptor to verify request  contains TE: chunked header? Just wanted to make sure we are testing the right path.

Reply:
added interceptor